### PR TITLE
Add software items to install/remove Plex Media Server, Emby Server and Docker

### PR DESCRIPTION
**Pull request by igorpecovnik**
# Description
Add software items to install/remove Plex Media Server, Emby Server and Docker

# Implementation Details

Add sub menu under Software, SW01.4 - Softy - 3rd party apps:
Add menu items with appropriate visibility:
SW22 - Install Plex Media server
SW23 - Remove Plex Media server
SW23 - Install Emby server
SW24 - Remove Emby server
SW25 - Install Docker Minimal
SW26 - Install Docker Engine 
SW27 - Remove Docker
SW28 - Purge all Docker images, containers, and volumes

Added function modules into config.ng.software.sh
install_plexmediaserver
install_embyserver
install_docker

- [x] Key changes introduced by this PR
- [x] Justification for the changes
- [x] Confirmation that no new external dependencies or modules have been introduced

# Documentation Summary

- [x] **Metadata Included:**  

- [x] **Document Generated:**  

# Testing Procedure
All new menu items tested successfully on:
- [x] Test 1: UEFIx86	 2GB 	Armbian 24.08.0 Bookworm
- [x] Test 2: Pine64 	 1GB 	Armbian 23.02.0 Focal
- [x] Test 3: Opizero+ 512MB 	Armbian 22.05.3 Jammy
- [x] Test 4: Rock-5C  32GB	Armbian 24.11.0 Noble 

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have ensured that my changes do not introduce new warnings or errors
- [x] No new external dependencies are included
- [x] Changes have been tested and verified
- [x] I have included necessary metadata in the code, including associative arrays

### CODE REVIEW

File: lib/armbian-configng/config.ng.jobs.json [hunk 1]
```diff
@@ -725,8 +725,120 @@
...
+                },
+                {
+                    "id": "SW01.4",
+                    "description": "Softy - 3rd party apps",
+                    "sub": [
+                            {
+                                "id": "SW21",
+                                "description": "Install Plex Media server",
+                                "command": [
+                                     "get_user_continue \"This operation will install Plex Media server.\n\nDo you wish to continue?\" process_input",
```
Comment (igorpecovnik) [inline lib/armbian-configng/config.ng.jobs.json:738]: Use prompt here https://github.com/armbian/configng/blob/main/lib/armbian-configng/config.ng.jobs.json#L23 and everywhere else down in the file.

File: lib/armbian-configng/config.ng.software.sh [hunk 2]
```diff
@@ -20,3 +20,95 @@ function see_monitoring() {
...
+install_plexmediaserver(){
+if [ ! -f /etc/apt/sources.list.d/plexmediaserver.list ]; then
+    echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/plexmediaserver.gpg] https://downloads.plex.tv/repo/deb public main" | sudo tee /etc/apt/sources.list.d/plexmediaserver.list > /dev/null 2>&1
+else
+    sed -i "/downloads.plex.tv/s/^#//g" /etc/apt/sources.list.d/plexmediaserver.list > /dev/null 2>&1
+fi
+# Note: for compatibility with existing source file in some builds format must be gpg not asc
+# and location must be /usr/share/keyrings
+wget -qO- https://downloads.plex.tv/plex-keys/PlexSign.key | gpg --dearmor | sudo tee /usr/share/keyrings/plexmediaserver.gpg > /dev/null 2>&1
+debconf-apt-progress -- apt-get update
```
Comment (igorpecovnik) [inline lib/armbian-configng/config.ng.software.sh:44]: Replace: 

`debconf-apt-progress --`
with
`apt_install_wrapper`

File: lib/armbian-configng/config.ng.software.sh [hunk 3]
```diff
@@ -20,3 +20,95 @@ function see_monitoring() {
...
+["update_skel,example"]="install_docker engine"
+["update_skel,status"]="Active"
+)
+#
+# Install Docker from repo using apt
+# Setup sources list and GPG key then install the app. If you want a full desktop then $1=desktop
+#
+install_docker(){
+# Check if repo for distribution exists.
+dist_id=$(cat /etc/*release | grep "ID=ubuntu\|ID=debian" | cut -c 4-)
```
Comment (igorpecovnik) [inline lib/armbian-configng/config.ng.software.sh:85]: We read this already when config starts.

Comment (schwar3kat) [reply lib/armbian-configng/config.ng.software.sh:85]: Found $DISTRO with capitalized format (Debian/Ubuntu).
I need debian/ubuntu.
Used distro=$(echo $DISTRO | tr '[:upper:]' '[:lower:]').
Is there a better way?

File: lib/armbian-configng/config.ng.jobs.json (hunk 1 above)
Comment (schwar3kat) [reply lib/armbian-configng/config.ng.jobs.json:737]: Used prompt for all of my menu items.
PR done for these minor tweaks.

File: lib/armbian-configng/config.ng.software.sh (hunk 2 above)
Comment (schwar3kat) [reply lib/armbian-configng/config.ng.software.sh:44]: Replaced.

**Pull request merged**